src/poliastro/twobody/orbit.py

```diff
@@ -578,25 +566,14 @@ def sample(self, values=None, method=mean_motion):
                 wrapped_nu = self.nu if self.nu < 180 * u.deg else self.nu - 360 * u.deg
                 nu_limit = max(np.arccos(-(1 - 1 / 3.0) / self.ecc), abs(wrapped_nu))
                 nu_values = np.linspace(-nu_limit, nu_limit, values)
+                time = self._generate_time_values(nu_values)
 
-            return self.sample(nu_values, method)
-
-        elif hasattr(values, "unit") and values.unit in ("rad", "deg"):
-            values = self._generate_time_values(values)
-
-        elif isinstance(values, time.Time):
-            values = values - self.epoch
-
-        elif isinstance(values, list):
-            # A list of Times is assumed
-            values = [(value - self.epoch).sec for value in values] * u.s
-
-        return self._sample(values, method)
+            return self._sample(time)
 
-    def _sample(self, time_values, method=mean_motion):
-        positions = method(self, time_values.to(u.s).value)
+    def _sample(self, time_values):
+        positions = [self.propagate(time).r.value for time in time_values]
```

Comment:
This change shouldn't be necessary if we follow this:

> The propagation methods always expect a list of times

Also, calling `self.propagate` in a loop would probably be very slow.